aspnetcore/tutorials/grpc/grpc-client.md

```diff
@@ -0,0 +1,224 @@
+
+Packages can be added with the following approaches:
+
+### [Visual Studio](#tab/visual-studio)
+
+* From the **Package Manager Console** window:
+  * Go to **View** > **Other Windows** > **Package Manager Console**
+  * Navigate to the directory in which the *GrpcGreeterClient.csproj* file exists
+  * Execute the following command:
+
+    ```powershell
+    Install-Package Grpc.Core
+    ```
+
+  * Repeat for Google.Protobuf and Grpc.Tools
+
+
+* From the **Manage NuGet Packages** dialog:
+  * Right-click the project in **Solution Explorer** > **Manage NuGet Packages**
+  * Set the **Package source** to "nuget.org"
+  * Enter "Grpc.Core" in the search box
+  * Select the "Grpc.Core" package from the **Browse** tab and click **Install**
+  * Repeat for Google.Protobuf and Grpc.Tools
+
+### [Visual Studio for Mac](#tab/visual-studio-mac)
+
+* Right-click the *Packages* folder in **Solution Pad** > **Add Packages...**
+* Set the **Add Packages** window's **Source** drop-down to "nuget.org"
+* Enter "Grpc.Core" in the search box
+* Select the "Grpc.Core" package from the results pane and click **Add Package**
+* Repeat for Google.Protobuf and Grpc.Tools
+
+### [Visual Studio Code](#tab/visual-studio-code)
+
+Run the following command from the **Integrated Terminal**:
+
+```console
+dotnet add TodoApi.csproj package Grpc.Core
+```
+
+Repeat for Google.Protobuf and Grpc.Tools
+
+### [.NET Core CLI](#tab/netcore-cli)
+
+Run the following command:
+
+```console
+dotnet add TodoApi.csproj package Grpc.Core
+```
+
+Repeat for Google.Protobuf and Grpc.Tools
+
+---
+
+## Add the greet.proto file
+
+Copy the **Protos\greet.proto** file from the gRPC Greeter service to the gRPC client project. Add the **greet.proto** file to the `<Protobuf>` item group of the GrpcGreeterClient project file:
+
+```XML
+<Protobuf Include="Protos\greet.proto" GrpcServices="Client" />
+```
+
+> [!NOTE]
+> You can open the project file of GrpcGreeterClient by right-clicking the project and selecting the **Edit GrpcGreeterClient.csproj** option from the dropdown menu.
+> ![new ASP.NET Core Web Application](grpc-start/_static/edit_csproj.png)
+> You can also navigate to the GrpcGreeterClient directory and edit the `GrpcGreeterClient.csproj` with your favorite editor.
+
+The `GrpcServices="Client"` attribute is added so that only the C# client assets are generated for the included protobuf file. Build the client project to trigger the generation of the C# client assets.
+
+## Create a GreeterClient and invoke the SayHello unary call
+
+Add the following code to `Main` method of the `Program.cs` file of the gRPC client project:
+
+[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcGreeterClient/Program.cs?name=snippet)]
+
+To access the required types the following using statements also need to be added:
+
+[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcGreeterClient/Program.cs?highlight=3-4)]
```

Comment:
This code sample is showing the entire file again, including `#region snippet`

Reply:
Gonna make a different snippet for the using section